src/Resources/Adjustments/Operations/CreateAdjustment.php

```diff
@@ -22,28 +22,46 @@ class CreateAdjustment implements \JsonSerializable
      */
     public function __construct(
         public readonly Action $action,
-        public readonly array|Undefined $items,
+        public readonly array|Undefined|null $items,
         public readonly string $reason,
         public readonly string $transactionId,
         public readonly AdjustmentType|Undefined $type = new Undefined(),
     ) {
-        if ($this->type === AdjustmentType::Partial() && ($this->items instanceof Undefined || empty($this->items))) {
+        $typeIsFull = AdjustmentType::Full()->equals($this->type);
+
+        if (! $typeIsFull && ($this->items instanceof Undefined || empty($this->items))) {
             throw InvalidArgumentException::arrayIsEmpty('items');
         }
+
+        if ($typeIsFull && is_array($this->items)) {
+            throw new InvalidArgumentException('items are not allowed when the adjustment type is full');
+        }
+    }
+
+    public static function full(Action $action, string $reason, string $transactionId): self
+    {
+        return new self($action, new Undefined(), $reason, $transactionId, AdjustmentType::Full());
+    }
+
+    public static function partial(Action $action, array $items, string $reason, string $transactionId): self
+    {
+        return new self($action, $items, $reason, $transactionId, AdjustmentType::Partial());
     }
 
     public function jsonSerialize(): array
     {
-        $items = [];
+        if (is_array($this->items)) {
+            $items = [];
```

Comment:
Moved `$items = []` inside `is_array($this->items)` check so that `Undefined` value is omitted from request